automation/src/test/scala/org/broadinstitute/dsp/consent/chains/DataSetChains.scala

```diff
@@ -0,0 +1,49 @@
+package org.broadinstitute.dsp.consent.chains
+
+import io.gatling.core.Predef._
+import io.gatling.core.structure.ChainBuilder
+import org.broadinstitute.dsp.consent.requests.Requests
+import spray.json._
+import DefaultJsonProtocol._
+import org.broadinstitute.dsp.consent.models.JsonProtocols
+import org.broadinstitute.dsp.consent.models.DataSetModels._
+import org.broadinstitute.dsp.consent.models.DataAccessRequestModels._
+import org.broadinstitute.dsp.consent.services._
+import scala.collection.mutable.ListBuffer
+
+object DataSetChains {
+    def dataSetCatalogPickTwo(additionalHeaders: Map[String, String]): ChainBuilder = {
+        exec(
+            Requests.User.dataSetCatalog(200, "${dacUserId}", additionalHeaders)
+        )
+        .exec { session =>
+            implicit val dataSetFormatJson = JsonProtocols.dataSetFormat
+            implicit val dataSetEntryJson = JsonProtocols.dataSetEntryFormat
+            implicit val dataAccessRequestJson = JsonProtocols.dataAccessRequestDraftFormat
+
+            var dataSetStr = session(Requests.DataSet.dataSetResponse).as[String]
+            val dataSets = dataSetStr.parseJson.convertTo[Seq[DataSet]]
+            
+            var chosenSets = dataSets
+                .filter(ds => ds.active && !ds.dataUse.collaboratorRequired.getOrElse(false))
+                .groupBy(_.dacId)
+                .map(_._2.head)
+                .take(2)
+
+            var setIds = new ListBuffer[Int]()
+            var entrySets = new ListBuffer[DataSetEntry]()
+            chosenSets.foreach(set => {
+                setIds += set.dataSetId
+                entrySets += DataSetEntryBuilder.fromDataSet(set)
+            })
```

Comment:
When you find yourself mutating vars in scala, there is almost always a more functional non-mutating way to do it. For example:
```
            val setIds: Seq[Int] = chosenSets.map(s => s.dataSetId).toSeq
            val entrySets: Seq[DataSetEntry] = chosenSets.map(s => DataSetEntryBuilder.fromDataSet(s)).toSeq
```
We strongly prefer immutable vals over their mutable counterparts.

Reply:
Thank you. Took this example and also updated another similar code block elsewhere.